app/src/main/kotlin/com/twain/interprep/utils/ValidationUtils.kt

```diff
@@ -1,19 +1,26 @@
 package com.twain.interprep.utils
 
 import com.twain.interprep.data.ui.TextInputAttributes
+import com.twain.interprep.data.ui.ValidationType
 
 /**
  * When shouldValidate is false, we don't need to validate the input text since user hasn't
  * edit the input yet.
  */
-fun isValidTextInput(
+fun isNotValidTextInput(
```

Comment:
Can we change the function to check and return true  if the text input is valid ?

Something like this :

fun isValidTextInput(
    shouldValidate: Boolean,
    text: String,
    attributes: TextInputAttributes,
    isError: Boolean
): Boolean {
    return if (shouldValidate) {
        !attributes.required || text.trim().isNotEmpty() // Valid if not required or not empty
        when (attributes.validationType) {
            ValidationType.REQUIRED -> { text.trim().isNotEmpty() }
            ValidationType.URL -> {
                text.isBlank() || isValidURL(text.trim()) // Valid if blank or a valid URL
            }
            ValidationType.NONE -> { true } // Always valid if validation type is NONE
        }
    } else {
        !isError // Valid if not in an error state
    }
}